src/Stratis.Bitcoin.Features.Consensus/Rules/PowConsensusRuleEngine.cs

```diff
@@ -54,5 +54,18 @@ public override Task Initialize()
         {
             return ((DBreezeCoinView)((CachedCoinView)this.UtxoSet).Inner).InitializeAsync();
         }
+
+        public override void Dispose()
+        {
+            var cache = this.UtxoSet as CachedCoinView;
+            if (cache != null)
+            {
+                this.logger.LogInformation("Flushing Cache CoinView...");
+                cache.FlushAsync().GetAwaiter().GetResult();
+                cache.Dispose();
+            }
+
+            ((DBreezeCoinView)((CachedCoinView)this.UtxoSet).Inner).Dispose();
+        }
```

Comment:
why here? this component does not seem to own those views

Reply:
I think it does, POW is supposedly a UTXO based chain, all interactions with UTXO happen in the rules mainly.

Reply:
In that case it should create the instance itself. But I don't believe that is the case. For example miner needs to access coin view, memory pool as well et cetera

Reply:
skype